lib/sync/default-dataHandlers.js

```diff
@@ -0,0 +1,79 @@
+var syncUtil = require('../sync-util');
+var util = require('util');
+
+var COLL_POSTFIX = '_collision';
+var mongo;
+
+module.exports = function (db) {
+
+  if (!db) {
+    throw new Error('MongoDB instance must be passed to module.');
+  }
+  mongo = db
+
+  return {
+    doList: function (dataset_id, params, meta_data, cb) {
+      syncUtil.doLog(dataset_id, 'verbose',
+                     'doList : ' + dataset_id +
+                     ' :: query_params=' + util.inspect(params) +
+                     ' :: meta_data=' + util.inspect(meta_data));
+      mongo.collection(dataset_id).find(params).toArray(cb);
```

Comment:
I think for the list operation, the sync framework expected the data return in a object, and the key should be the uid of each record. See https://access.redhat.com/documentation/en-us/red_hat_mobile_application_platform_hosted/3/html/cloud_api/fh-sync#fh-sync-handlelist